app/src/main/java/chat/rocket/android/chatrooms/ui/ChatRoomsFragment.kt

```diff
@@ -393,15 +390,19 @@ class ChatRoomsFragment : Fragment(), ChatRoomsView {
         }
 
         searchCloseButton?.setOnClickListener { v ->
-            searchView?.clearFocus()
-            searchView?.setQuery("", false)
-            searchCloseButton?.setImageResource(0)
-            viewModel.showLastMessage = true
+            clearSearch()
         }
 
         searchView?.onQueryTextListener { queryChatRoomsByName(it) }
     }
 
+    private fun clearSearch() {
+        searchView?.clearFocus()
+        searchView?.setQuery("", false)
+        searchCloseButton?.setImageResource(0)
+        viewModel.showLastMessage = true
+    }
+
```

Comment:
onPause() and onResume() don't need to do all 4 of the actions in clearSearch(). Is it definitely ok/right to have all of them happen at those times anyways?

Reply:
These 4 actions completely clear the search in all cases.  It does not hurt in any way, that I am aware of, to be performing these actions, even in the case that it will not result in a change of state.